### DISCUSSION TITLE
网络差的情况下网络暴涨，4M码率的码流网络带宽占用可以翻10倍

### DISCUSSION THREAD

**Discussion by BigBigGrayWolf** [Tencent/tquic#373] [Q&A (问题和回答)]
在Linux上用tc qdisc add dev eth0 root netem delay 30ms 150ms loss 5%模拟弱网。不限带宽使用copa算法（其他算法测试抗网络抖动效果没有copa好）的情况下带宽在20-50M之间波动。可以通过什么参数调整一下。在带宽增长不明显的情况下播放延时在1秒内吗？

**Comment by ChrisTeg** [comment]
『4M码率的码流网络带宽占用可以翻10倍』是否指码率为4Mbps，但估算出来的pacing rate在20Mbps~50Mbps，而这会影响播放延时？如果码率为4Mbps，但估算的pacing rate远大于该码率，那传输会受限于实际码率，进入app-limited状态，大pacing rate的影响就比较小了。
- 能否具体描述一下你的问题和场景？如果有日志文件也麻烦贴出。
- Copa的本质是通过排队延迟推算一个理想的pacing rate，并通过调节窗口来趋近这个rate。Copa算法可以通过调节 $$\delta$$ 系数（0-1）来调节target rate及与buffer-filling流的竞争能力，理论上， $$\delta$$ 越大，排队越小，时延越小。目前 $$\delta$$ 尚未开放配置，若有需要，可以先通过修改COPA_DELTA常量进行测试。

https://github.com/Tencent/tquic/blob/96787725c03376fd00cbf7a47be1f7680e919ef5/src/congestion_control/copa.rs#L39
